core/src/main/scala/shapeless/generic.scala

```diff
@@ -380,82 +381,85 @@ trait CaseClassMacros extends ReprTypes {
   }
 
   def ctorsOfAux(tpe: Type, hk: Boolean): List[Type] = {
-    def collectCtors(classSym: ClassSymbol): List[ClassSymbol] = {
-      classSym.knownDirectSubclasses.toList flatMap { child0 =>
+    def collectCtors(classSym: ClassSymbol): List[ClassSymbol] =
+      classSym.knownDirectSubclasses.toList.flatMap { child0 =>
         val child = child0.asClass
         child.typeSignature // Workaround for <https://issues.scala-lang.org/browse/SI-7755>
-        if (isCaseClassLike(child) || isCaseObjectLike(child))
-          List(child)
-        else if (child.isSealed)
-          collectCtors(child)
-        else
-          abort(s"$child is not case class like or a sealed trait")
+        if (isCaseClassLike(child) || isCaseObjectLike(child)) List(child)
+        else if (child.isSealed) collectCtors(child)
+        else abort(s"$child is not case class like or a sealed trait")
       }
-    }
 
-    if(isProduct(tpe))
+    if (isProduct(tpe)) {
       List(tpe)
-    else if(isCoproduct(tpe)) {
+    } else if (isCoproduct(tpe)) {
       val basePre = prefix(tpe)
       val baseSym = classSym(tpe)
-      val baseTpe =
-        if(!hk) tpe
-        else {
-          val tc = tpe.typeConstructor
-          val paramSym = tc.typeParams.head
-          val paramTpe = paramSym.asType.toType
-          appliedType(tc, paramTpe)
-        }
+      val baseTpe = if (!hk) tpe else {
+        val tc = tpe.typeConstructor
+        appliedType(tc, tc.typeParams.head.asType.toType)
+      }
```

Comment:
Similarly with the removal of the intermediate val ... I'm not opposed to the change in principle, but it is making it harder for me to review this PR.